### ISSUE TITLE
[MPS] BatchNorm3d crashes with NSRangeException on non-contiguous tensors

### ISSUE THREAD

**Issue by helomelo1** [pytorch/pytorch#190645]
### 🐛 Describe the bug

`nn.BatchNorm3d` on the `mps` device hard-crashes with an Objective-C `NSRangeException` when the input tensor is **non-contiguous** (e.g., after a `.permute()`).

The same code works on `cpu`. It also works on `mps` if `.contiguous()` is called before passing the tensor to `BatchNorm3d`.

This suggests the MPS implementation of BatchNorm3d does not correctly handle this valid non-contiguous tensor layout.

### Code to reproduce

```python
import torch
import torch.nn as nn

device = torch.device("mps")

# Create a 5D tensor and make it non-contiguous via permute
x = torch.randn(64, 150, 10, 16, 51, device=device).permute(0, 4, 1, 2, 3)
print(f"Is contiguous? {x.is_contiguous()}")  # False

bn = nn.BatchNorm3d(51).to(device)

# This crashes with NSRangeException
output = bn(x)
```

### Workaround

Adding `.contiguous()` before the BatchNorm call prevents the crash:

```python
x = x.contiguous()  # this single line fixes it
output = bn(x)
```

### Error traceback

```
*** Terminating app due to uncaught exception 'NSRangeException', reason:
'*** -[__NSArrayM objectAtIndexedSubscript:]: index 4 beyond bounds for empty array'

libtorch_cpu.dylib: batch_norm_mps_out → batch_norm_mps → native_batch_norm → batch_norm
libc++abi: terminating due to uncaught exception of type NSException
zsh: abort
```

### Expected behavior

The MPS backend should either handle non-contiguous tensors in `BatchNorm3d` (like CPU/CUDA do), or internally call `.contiguous()` before processing.

### Versions

PyTorch version: 2.9.1
Is debug build: False
CUDA used to build PyTorch: None
ROCM used to build PyTorch: N/A

OS: macOS 26.5.2 (arm64)
GCC version: Could not collect
Clang version: 21.0.0 (clang-2100.1.1.101)
CMake version: Could not collect
Libc version: N/A

Python version: 3.14.2 (v3.14.2:df793163d58, Dec  5 2025, 12:18:06) [Clang 16.0.0 (clang-1600.0.26.6)] (64-bit runtime)
Python platform: macOS-26.5.2-arm64-arm-64bit-Mach-O
Is CUDA available: False
CUDA runtime version: No CUDA
CUDA_MODULE_LOADING set to: N/A
GPU models and configuration: No CUDA
Nvidia driver version: No CUDA
cuDNN version: No CUDA
Is XPU available: False
HIP runtime version: N/A
MIOpen runtime version: N/A
Is XNNPACK available: True
Caching allocator config: N/A

CPU:
Apple M5

Versions of relevant libraries:
[pip3] numpy==2.4.0
[pip3] onnx==1.21.0
[pip3] onnx-ir==0.2.0
[pip3] onnxscript==0.6.2
[pip3] torch==2.9.1
[pip3] torch-geometric==2.7.0
[pip3] torchvision==0.24.1
[conda] Could not collect

cc @kulinseth @malfet @DenisVieriu97 @jhavukainen @aditvenk @Isalia20

**Comment by Isalia20** [comment]
This is fixed on `torch==2.13.0`. Can you please check it?

**Comment by helomelo1** [comment]
I just upgraded to 2.13.0 and can confirm it works perfectly now. Thanks for the quick response! Closing this.
